hazelcast-jet-core/src/main/java/com/hazelcast/jet/Edge.java

```diff
@@ -317,7 +317,35 @@ public Edge setConfig(EdgeConfig config) {
 
     @Override
     public String toString() {
-        return '(' + sourceName + ", " + sourceOrdinal + ") -> (" + destName + ", " + destOrdinal + ')';
+        final StringBuilder b = new StringBuilder();
+        if (sourceOrdinal == 0 && destOrdinal == 0) {
+            b.append("between(").append(sourceName).append(", ").append(destName).append(')');
+        } else {
+            b.append("from(").append(sourceName);
+            if (sourceOrdinal != 0) {
+                b.append(", ").append(sourceOrdinal);
+            }
+            b.append(").to(").append(destName);
+            if (destOrdinal != 0) {
+                b.append(", ").append(destOrdinal);
+            }
+            b.append(')');
+        }
+        switch (getForwardingPattern()) {
+            case VARIABLE_UNICAST:
+                break;
+            case PARTITIONED:
+                b.append(getPartitioner() instanceof Single ? ".allToOne()" : ".partitioned(?)");
```

Comment:
1. Ellipsis (...) means, that there is something to continue. We know, what it is, but just don't write it here.
2. I can easily distinguish between "." and "..." :). Notice the usage of quotes :).